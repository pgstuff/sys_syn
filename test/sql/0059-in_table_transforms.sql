BEGIN;

CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text varchar(255),
        test_table_date date,
        test_table_datetime timestamp with time zone,
        test_table_us_eastern_datetime timestamp without time zone,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));

INSERT INTO sys_syn.in_table_transforms(
        rule_group_id,          priority,       final_ids,              in_group_id_like,       in_table_id_like,
        new_in_partition_count)
VALUES ('in_partition_count',   50,             '{}',                   '%',                    null,
        4);

INSERT INTO sys_syn.in_groups_def
        (in_group_id,   parent_in_group_id,     rule_group_ids)
VALUES  ('in',          NULL,                   ARRAY['in_partition_count']),
        ('in2',         'in',                   NULL),
        ('in3',         'in2',                  ARRAY['in_partition_count']);

SELECT sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in3');

INSERT INTO sys_syn.in_table_transforms(
        rule_group_id,          priority,       final_ids,              in_group_id_like,       schema_like,    in_table_id_like,
        omit)
VALUES (null,                   20,             '{}',                   null,                   'user_%',       null,
        true);

SELECT sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in', 'user_data');

ROLLBACK;
