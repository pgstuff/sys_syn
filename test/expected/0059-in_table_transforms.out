BEGIN;
CREATE EXTENSION tinyint SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text varchar(255),
        test_table_date date,
        test_table_datetime timestamp with time zone,
        test_table_us_eastern_datetime timestamp without time zone,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_table_transforms(
        rule_group_id,          priority,       final_ids,              in_group_id_like,       in_table_id_like,
        new_in_partition_count)
VALUES ('in_partition_count',   50,             '{}',                   '%',                    null,
        4);
INSERT INTO sys_syn.in_groups_def
        (in_group_id,   parent_in_group_id,     rule_group_ids)
VALUES  ('in',          NULL,                   ARRAY['in_partition_count']),
        ('in2',         'in',                   NULL),
        ('in3',         'in2',                  ARRAY['in_partition_count']);
SELECT sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in3');
                                                                     in_table_create_sql                                                                     
-------------------------------------------------------------------------------------------------------------------------------------------------------------
 SELECT  sys_syn.in_table_create(                                                                                                                           +
                 schema          => 'user_data'::regnamespace,                                                                                              +
                 in_table_id     => 'test_table',                                                                                                           +
                 in_group_id     => 'in3',                                                                                                                  +
                 in_pull_id      => NULL,                                                                                                                   +
                 in_columns      => ARRAY[                                                                                                                  +
                        $COL$("test_table_id","integer",Id,"in_source.test_table_id",,,,,)$COL$,                                                            +
                        $COL$("test_table_text","character varying(255)",Attribute,"in_source.test_table_text",,,,,)$COL$,                                  +
                        $COL$("test_table_date","date",Attribute,"in_source.test_table_date",,,,,)$COL$,                                                    +
                        $COL$("test_table_datetime","timestamp with time zone",Attribute,"in_source.test_table_datetime",,,,,)$COL$,                        +
                        $COL$("test_table_us_eastern_datetime","timestamp without time zone",Attribute,"in_source.test_table_us_eastern_datetime",,,,,)$COL$+
                 ]::sys_syn.create_in_column[],                                                                                                             +
                 full_table_reference    => 'user_data.test_table',                                                                                         +
                 changes_table_reference => NULL,                                                                                                           +
                 full_sql                => NULL,                                                                                                           +
                 changes_sql             => NULL,                                                                                                           +
                 full_pre_sql            => NULL,                                                                                                           +
                 changes_pre_sql         => NULL,                                                                                                           +
                 full_post_sql           => NULL,                                                                                                           +
                 changes_post_sql        => NULL,                                                                                                           +
                 enable_deletes_implied  => 'true',                                                                                                         +
                 null_key_handler        => 'none'::sys_syn.null_key_handler,                                                                               +
                 key_violation_handler   => 'none'::sys_syn.key_violation_handler,                                                                          +
                 full_prepull_id         => NULL,                                                                                                           +
                 changes_prepull_id      => NULL,                                                                                                           +
                 record_comparison_different=>NULL,                                                                                                         +
                 record_comparison_same  => NULL,                                                                                                           +
                 tablespace              => NULL,                                                                                                           +
                 in_partitions           => ARRAY[                                                                                                          +
                         $PART$("",)$PART$,                                                                                                                 +
                         $PART$("",)$PART$,                                                                                                                 +
                         $PART$("",)$PART$,                                                                                                                 +
                         $PART$("",)$PART$]::sys_syn.create_in_partition[]                                                                                  +
         );
(1 row)

INSERT INTO sys_syn.in_table_transforms(
        rule_group_id,          priority,       final_ids,              in_group_id_like,       schema_like,    in_table_id_like,
        omit)
VALUES (null,                   20,             '{}',                   null,                   'user_%',       null,
        true);
SELECT sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in', 'user_data');
 in_table_create_sql 
---------------------
 
(1 row)

ROLLBACK;
